### PR TITLE
[gatsby-source-nacelle] Enable Contentful Preview

### PR DESCRIPTION
**Pull request by NWRichmond**
**Story:** [DRMS-801](https://nacelle.atlassian.net/browse/DRMS-801) 

> CMS Preview Functionality

### Type of Change

- [x] Non-breaking feature
- [x] Chore (docs, refactor, etc)

#### Content Preview

- Enables content preview from Contentful
- In preview mode, the data shape and associated GraphQL queries are consistent with non-preview / production mode
- Documented in `README`
  - TODO: add documentation to Nacelle's public-facing docs

#### Refactor

- Break down code into modular `services/`, `utils/`, etc. to improve readability of `gatsby-node.js`
- Rely on Gatsby's new [`pluginOptionsSchema`](https://www.gatsbyjs.com/docs/node-apis/#pluginOptionsSchema) API instead of homegrown checks and sanitizing of plugin options provided by the user

### How to Test

- `npm run bootstrap` from monorepo root
-  add environment variables to `examples/gatsby/.env` (I can provide these in Slack)
- `npm run dev` to start the Gatsby dev server
- open the GraphiQL Explorer at `http://localhost:8000/___graphql`
- query a piece of `NacelleContent`
- make a change to that piece of content [in Contentful](https://app.contentful.com/spaces/iojm91u4ez5c/entries?id=dDmj499Hk1E8dLQz&order.fieldId=updatedAt&order.direction=descending&displayedFieldIds=contentType&displayedFieldIds=updatedAt&displayedFieldIds=author)
- hit the **Refresh Content** button in the GraphiQL Explorer and execute the query again to preview changes made in Contentful

### CODE REVIEW

File: packages/gatsby-source-nacelle/package.json [hunk 1]
```diff
@@ -21,16 +21,19 @@
...
+    "lodash.isobject": "^3.0.2",
+    "lodash.transform": "^4.6.0"
   },
   "peerDependencies": {
     "gatsby": "^2.0.32"
   },
   "devDependencies": {
-    "eslint": "^7.6.0",
-    "lint-staged": "^10.2.11"
+    "eslint": "^7.13.0",
```
Comment (johnstonbl01) [inline packages/gatsby-source-nacelle/package.json:35]: This ESLint version is > the version in the lerna package.json (which gets hoisted). This should be the same version that's in the root of the project (7.6.0)

File: packages/gatsby-source-nacelle/src/source-nodes/contentful-preview.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  try {
+    const contentData = await client.data
+      .allContent()
+      .then((
+        content // change name of reserved keys
+      ) => replaceKeys(content, { id: 'remoteId', fields: 'remoteFields' }))
+      .catch((err) => {
+        throw new Error(
+          `Problem fetching content:\n\n${JSON.stringify(err, null, 2)}`
```
Comment (johnstonbl01) [inline packages/gatsby-source-nacelle/src/source-nodes/contentful-preview.js:31]: Won't this error just get caught by the `try... catch`? If so, we'd end up with `Problem sourcing content from Contentful Preview API: [object object]` or something of that sort. What about something like:

```js
const contentData = await client.data.allContent();
const formattedData = replaceKeys(content, { id: 'remoteId', fields: 'remoteFields' });

contentData.forEach(...)
```

```js
...
} catch (err) {
  console.error(`Problem sourcing content from Contentful Preview API: ${err.message}`);
}
```
```

**Review by johnstonbl01 — CHANGES_REQUESTED** [review]

File: packages/gatsby-source-nacelle/package.json (hunk 1 above)
Comment (NWRichmond) [reply packages/gatsby-source-nacelle/package.json:35]: No problem! I'll change it back to `7.6.0`.

File: packages/gatsby-source-nacelle/src/source-nodes/contentful-preview.js (hunk 2 above)
Comment (NWRichmond) [reply packages/gatsby-source-nacelle/src/source-nodes/contentful-preview.js]: Right! Changed :)

**Review by johnstonbl01 — APPROVED** [review]

**Pull request merged**